### PR TITLE
Mark some single-argument constructors explicit

### PR DESCRIPTION
**Pull request by JoeLoser**
Requested reviewers:
@mellery451 @seelabs

### CODE REVIEW

File: src/ripple/app/ledger/Ledger.cpp [hunk 1]
```diff
@@ -87,7 +87,7 @@ class Ledger::sles_iter_impl
 
     sles_iter_impl (sles_iter_impl const&) = default;
 
-    sles_iter_impl (SHAMap::const_iterator iter)
+    explicit sles_iter_impl (SHAMap::const_iterator iter)
```
Comment (seelabs) [inline src/ripple/app/ledger/Ledger.cpp]: I'd keep this one as-is. That seems like a reasonable automatic conversion to allow.

**Comment by seelabs** [comment]
I left one comment about allowing one of the conversions, but the rest look good to me. Thanks!

**Comment by JoeLoser** [comment]
Thanks for the review @seelabs. Just rebased and force-pushed taking into account your feedback since it was only a one-word change.

**Comment by mellery451** [comment]
in f658656b8

**Pull request closed without merging**